### ISSUE TITLE
Key Released event is either not properly ordered or missing when fps is limited

### ISSUE THREAD

**Issue by Neko-Box-Coder** [SFML/SFML#2191]
## Subject of the issue

Key Released event is either not properly ordered or missing when fps is limited, creating a "key stuck" effect.

## Your environment

* Your OS / distro / window manager used: 
  * Manajro KDE 5.18.5
* Your version of SFML (2.5.0, git master, etc): 
  * Tested on latest snapshot, but it also existed on snapshots from April. Very likely that it has existed for quite a while.
* Your compiler and compiler version used: 
  * GCC 12
* Special compiler flags used: 
```cmake
# relevant cmake commands used:
set (CMAKE_CXX_STANDARD 11)
#Find shared libraries in the same location as the executable
SET(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-rpath -Wl,$ORIGIN")
set(CMAKE_CXX_FLAGS "-fpermissive -std=c++0x")  #For surpressing goto warning
```
## Steps to reproduce
1. Limit the framerate, either using built-in set frame limit method or manually set the thread to sleep.
2. Hold any key for a few seconds and release, in this case I am using the "a" key.  

Tell us how to reproduce this issue. Please provide a [minimal, complete and verifiable example](https://stackoverflow.com/help/mcve), you can use the follow template as a start:

```cpp
#include <iostream> 
#include "SFML/Graphics.hpp"
#include "SFML/Window.hpp"
#include <thread>

int main()
{
    sf::RenderWindow window(sf::VideoMode(sf::Vector2u(500, 500)), "SFML works!");

    window.setFramerateLimit(30);

    bool pressed = false;
    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if(event.type == sf::Event::KeyPressed)
            {
                std::cout<<("key pressed\n");

                if(event.key.code == sf::Keyboard::Key::A)
                    pressed = true;

                continue;
            }
            else if(event.type == sf::Event::KeyReleased)
            {
                std::cout<<("key released\n");

                if(event.key.code == sf::Keyboard::Key::A)
                    pressed = false;
                continue;
            }

            if (event.type == sf::Event::Closed)
                window.close();
        }

        if(pressed)
            window.clear(sf::Color(255, 0, 0, 255));
        else
            window.clear(sf::Color(255, 255, 255, 255));

        window.display();
        // std::this_thread::sleep_for(std::chrono::milliseconds(20));
    }

    return 0;
}
```

## Expected behavior

The release key event should always come after the key pressed event. 
With the example I have given, the background should always be white unless the "a" key is being held.

## Actual behavior

The release key event comes before the last key pressed event. 
With the example I have given, the background gets "stuck" with the red background occasionally.

![Peek 2022-09-03 18-30](https://user-images.githubusercontent.com/93885501/188282238-18d5b067-7b27-49c7-9d6d-54dcc11230eb.gif)

**Comment by kimci86** [comment]
This behavior was introduced by mistake in #1850 (see https://github.com/SFML/SFML/pull/1850#issuecomment-989994414).
There is currently an open PR #1894 trying to solve it, without much success for now.

**Comment by Neko-Box-Coder** [comment]
Thanks for informing me. 

I am using fcitx5 instead of ibus, so this is also affecting fcitx5 if this makes any difference.
I have also tried limiting at 60 and 90 fps but although less frequent, the problem still exists. 

Do you know if there's any plan to fix this?
Given this is probably affecting most use cases with any sort of framerate locking.

Thanks.

**Comment by eXpl0it3r** [comment]
Fixed with #2242

If anyone runs into an issue, please let us know 🙂
